test/SerialIOTest.cpp

```diff
@@ -4427,17 +4427,23 @@ TEST_CASE( "automatically_deactivate_span", "[serial][adios2]" )
     }
 
     // enable span-based API indiscriminately
-    try
     {
         std::string enable = R"END(
 {
   "adios2": {
     "use_span_based_put": true
   }
 })END";
-        Series write( "../samples/span_based.bp", Access::CREATE, enable );
-        auto E_uncompressed = write.iterations[ 0 ].meshes[ "E" ][ "x" ];
-        auto E_compressed = write.iterations[ 0 ].meshes[ "E" ][ "y" ];
+        /*
+         * Use a pointer so we can leak the memory later.
```

Comment:
From the change I cannot see this immediately: Why is this change now necessary and wasn't before?
Due to the second point in the PR description
> ADIOS2 now throws an error if using the span-based Put() API in combination with operators, so we must adapt our tests

cc @williamfgc?
```suggestion
        /* Use a pointer so we can leak the memory later.
```

> Destructing should always work, even if errors are thrown. We cannot do much against segfaults in our flushing routines during destruction, but we should check if this can be solved without leaking the memory.

Agreed.

> Should we add an API call `Series::invalidate()` to indicate that this `Series` object should perform no further operations (especially: only free memory during destruction, do nothing else, no flushing)?

Possible for the state of a `Series`, but that means we need to check on every member function access if the class is in valid state (and so for every other object down the dependency tree)...

Reply:
> Possible for the state of a Series, but that means we need to check on every member function access if the class is in valid state (and so for every other object down the dependency tree)...

Not necessarily? I would propose the following semantics for `Series::invalidate()`:
* Any access to an invalidated Series is undefined behavior
* Destruction must work

To be fair, these (repeatedly occuring) issues make me increasingly wary of putting too much logic into destructors. Sometimes, you just want to cleanly get rid of allocated memory and `invalidate()` would essentially indicate this.

Reply:
Also, in case that you don't want to introduce this sort of undefined behavior, we can put a check on `Series::get()` which nearly every method in the Series API calls at some time.

But in my opinion, if calling something like `Series::invalidate()`, you are explicitly asking for that kind of situation, so I don't see a big problem with undefined behavior here

Reply:
> From the change I cannot see this immediately: Why is this change now necessary and wasn't before?
> Due to the second point in the PR description

Ah, I missed this part. See [this PR](https://github.com/ornladios/ADIOS2/pull/2981) and the [corresponding issue](https://github.com/ornladios/ADIOS2/issues/2965).
Previously, doing this just silently created invalid datasets and it took a lot of time to find out why they were invalid, so there is now a clear error message for that situation in ADIOS2.

Reply:
@ax3l @franzpoeschel let us know if there is more of these. As much as adios2 throws exceptions, they is always a missing case. Thanks!